### ISSUE TITLE
Use new-style asyncio (python 3.8 DeprecationWarning)

### ISSUE THREAD

**Issue by jquast** [jquast/telnetlib3#46]
This library started a long time ago, many of the patterns used are now marked deprecated in 3.8 and will be removed in future python versions

```
telnetlib3/stream_reader.py:13
  /Users/jq/Code/telnetlib3/telnetlib3/stream_reader.py:13: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def readline(self):

telnetlib3/stream_reader.py:136
  /Users/jq/Code/telnetlib3/telnetlib3/stream_reader.py:136: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def readline(self):

telnetlib3/stream_reader.py:148
  /Users/jq/Code/telnetlib3/telnetlib3/stream_reader.py:148: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def read(self, n=-1):

telnetlib3/stream_reader.py:198
  /Users/jq/Code/telnetlib3/telnetlib3/stream_reader.py:198: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def readexactly(self, n):

telnetlib3/server_shell.py:12
  /Users/jq/Code/telnetlib3/telnetlib3/server_shell.py:12: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def telnet_server_shell(reader, writer):

telnetlib3/server_shell.py:64
  /Users/jq/Code/telnetlib3/telnetlib3/server_shell.py:64: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def readline(reader, writer):

telnetlib3/server.py:379
  /Users/jq/Code/telnetlib3/telnetlib3/server.py:379: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def create_server(host=None, port=23, protocol_factory=TelnetServer, **kwds):

telnetlib3/server.py:442
  /Users/jq/Code/telnetlib3/telnetlib3/server.py:442: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def _sigterm_handler(server, log):

telnetlib3/client_shell.py:125
  /Users/jq/Code/telnetlib3/telnetlib3/client_shell.py:125: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def make_stdio(self):

telnetlib3/client_shell.py:160
  /Users/jq/Code/telnetlib3/telnetlib3/client_shell.py:160: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def telnet_client_shell(telnet_reader, telnet_writer):

telnetlib3/client.py:239
  /Users/jq/Code/telnetlib3/telnetlib3/client.py:239: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def open_connection(host=None, port=23, *, client_factory=None, loop=None,

telnetlib3/tests/test_charset.py::test_telnet_server_on_charset[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_charset.py:39: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_create_server_on_connect[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:74: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_open_close[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:94: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_advanced_negotiation[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:160: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_closed_by_client[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:196: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_eof_by_client[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:224: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_closed_by_server[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:251: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_idle_duration[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:291: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_closed_by_error[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:344: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_negotiation_fail[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:393: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_server_cmdline[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:464: DeprecationWarning: The loop argument is deprecated since Python 3.8 and scheduled for removal in Python 3.10.
    proc = await asyncio.create_subprocess_exec(

telnetlib3/tests/test_core.py::test_telnet_server_cmdline[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:476: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_core.py::test_telnet_client_cmdline[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:535: DeprecationWarning: The loop argument is deprecated since Python 3.8 and scheduled for removal in Python 3.10.
    proc = await asyncio.create_subprocess_exec(

telnetlib3/tests/test_core.py::test_telnet_client_tty_cmdline[127.0.0.1]
  /Users/jq/Code/telnetlib3/.tox/py38/lib/python3.8/site-packages/pexpect/_async.py:8: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def expect_async(expecter, timeout=None):

telnetlib3/tests/test_core.py::test_telnet_client_tty_cmdline[127.0.0.1]
  /Users/jq/Code/telnetlib3/.tox/py38/lib/python3.8/site-packages/pexpect/_async.py:31: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def repl_run_command_async(repl, cmdlines, timeout=-1):

telnetlib3/tests/test_core.py::test_telnet_client_cmdline_stdin_pipe[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:593: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_core.py::test_telnet_client_cmdline_stdin_pipe[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:609: DeprecationWarning: The loop argument is deprecated since Python 3.8 and scheduled for removal in Python 3.10.
    proc = await asyncio.create_subprocess_exec(

telnetlib3/tests/test_core.py::test_telnet_server_as_module
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:446: DeprecationWarning: The loop argument is deprecated since Python 3.8 and scheduled for removal in Python 3.10.
    proc = await asyncio.create_subprocess_exec(

telnetlib3/tests/test_core.py::test_telnet_client_as_module
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_core.py:504: DeprecationWarning: The loop argument is deprecated since Python 3.8 and scheduled for removal in Python 3.10.
    proc = await asyncio.create_subprocess_exec(

telnetlib3/tests/test_encoding.py::test_telnet_server_encoding_default[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:31: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_encoding.py::test_telnet_server_encoding_client_will[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:85: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_encoding.py::test_telnet_server_encoding_server_do[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:112: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_encoding.py::test_telnet_server_encoding_bidirectional[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:139: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_encoding.py::test_telnet_server_encoding_by_LANG[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:194: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_encoding.py::test_telnet_server_binary_mode[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:224: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def binary_shell(reader, writer):

telnetlib3/tests/test_encoding.py::test_telnet_server_binary_mode[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_encoding.py:241: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_environ.py::test_telnet_server_on_environ[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_environ.py:38: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_linemode.py::test_server_demands_remote_linemode_client_agrees[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_linemode.py:37: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_linemode.py::test_server_demands_remote_linemode_client_demands_local[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_linemode.py:91: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_naws.py::test_telnet_server_on_naws[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_naws.py:38: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_shell.py::test_telnet_server_shell_as_coroutine[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:33: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_shell.py::test_telnet_server_shell_as_coroutine[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:48: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_shell.py::test_telnet_client_shell_as_coroutine[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:82: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_shell.py::test_telnet_server_shell_make_coro_by_function[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:112: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_shell.py::test_telnet_server_no_shell[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:136: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_shell.py::test_telnet_server_given_shell[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:162: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_shell.py::test_telnet_server_shell_eof[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_shell.py:305: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_timeout.py::test_telnet_server_default_timeout[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_timeout.py:33: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_timeout.py::test_telnet_server_set_timeout[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_timeout.py:60: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_timeout.py::test_telnet_server_waitfor_timeout[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_timeout.py:89: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_tspeed.py::test_telnet_server_on_tspeed[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_tspeed.py:35: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:35: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype_beyond_max[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:79: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype_empty[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:124: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype_looped[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:164: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype_repeated[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:206: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_ttype.py::test_telnet_server_on_ttype_mud[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_ttype.py:248: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_iac_do_twice_replies_once[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:125: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_iac_do_twice_replies_once[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:136: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_iac_dont_dont[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:154: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_iac_dont_dont[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:165: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_slc_simul[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:223: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_slc_simul[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:254: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_unhandled_do_sends_wont[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:284: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_writelines_bytes[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:301: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_writelines_bytes[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:310: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_writelines_unicode[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:326: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_writelines_unicode[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:335: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_send_ga[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:361: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_send_ga[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:370: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_not_send_ga[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:390: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_not_send_ga[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:399: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_not_send_eor[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:415: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_not_send_eor[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:424: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_writer.py::test_send_eor[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:445: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def shell(reader, writer):

telnetlib3/tests/test_writer.py::test_send_eor[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_writer.py:456: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    client_reader, client_writer = await asyncio.open_connection(

telnetlib3/tests/test_xdisploc.py::test_telnet_server_on_xdisploc[127.0.0.1]
  /Users/jq/Code/telnetlib3/telnetlib3/tests/test_xdisploc.py:39: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    reader, writer = await asyncio.open_connection(
```

**Comment by JPHutchins** [comment]
This would be nice - AFAIK this is a matter of replacing the decorators with async keyword?

**Comment by jquast** [comment]
Yes

**Comment by tarkatronic** [comment]
@jquast Is this something you've started working on? I forked this library over to https://github.com/Facet-MUD-Project/telnetlib3 so that I could try taking a poke at it, as a part of better familiarizing myself with the modern asyncio.

As should be expected, it's turned into quite a beast of a change. And my work has somewhat stalled, because... well... free time.

**Comment by jquast** [comment]
Any next major release will have to address this, I have more important (more pypi downloads) work to finish in wcwidth first, PR's welcome, best wishes

**Comment by tarkatronic** [comment]
Cool, then I shall continue my work and hope I don't botch it too badly! Worst case I'll just... not submit a PR, I suppose.

One question came to mind during this work: Would you be open to a PR to format the code with `black`? I've gotten so used to automatic formatting that the manual formatting keeps throwing me for a loop.

**Comment by jquast** [comment]
Sure, go ahead and black it

**Comment by shaozi** [comment]
Any update on this pull?

**Comment by jquast** [comment]
It's in progress, https://github.com/jquast/telnetlib3/pull/58

There is one failing test remaining. It was a test added to detect a cyclic reference. The cyclic reference is from a hack used for testing, I'd like to add a more useful public API to await a negotiation state, like`wait_for_negotiation({ECHO: True, SGA: True})` to remove the need for this hack

**Comment by jonathanslenders** [comment]
Just FYI, telnetlib3 is now broken on Python 3.11:

```python
  File ".../site-packages/telnetlib3/stream_reader.py", line 36, in TelnetReader
    @asyncio.coroutine
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'. Did you mean: 'coroutines'?
```

**Comment by jquast** [comment]
this is fixed in 2.0.0 released to pypi today

**Comment by jonathanslenders** [comment]
Thank you @jquast, for all the work!
